distributed/client.py

```diff
@@ -2381,6 +2381,58 @@ def collections_to_dsk(collections, optimize_graph=True, **kwargs):
 
         return dsk
 
+    @gen.coroutine
+    def _get_futures_error(self, future):
+        """
+        Ask the scheduler details of the sub-task of the given failed future
+
+        Parameters
+        ----------
+        future: future that failed
+
+        Returns
+        -------
+        The function that failed, its arguments, and the keys that it depends
+        on.
+        """
+        futures = futures_of(future)
+        deps, cause, task = yield self.scheduler.cause_of_failure(
+                keys=[f.key for f in futures])
+        function, args, kwargs = Worker._deserialize(None, **task)
+        raise gen.Return((function, args, kwargs, deps))
+
+    @gen.coroutine
+    def _get_specific_keys(self, keys):
+        # Inform scheduler that we want these keys.
+        futures = self._graph_to_futures({}, keys)
+
+        # Get data from remote workers, pack into args and kwargs
+        raise gen.Return(self._gather(futures))
```

Comment:
This doesn't need to be a coroutine.  It never yields.

It might also be enough to replace calls to `_get_specific_keys` with just the graph_to_futures and _gather calls directly?  I think that there is a cost to adding a new method.  Future readers will have to backtrack to find what it does.  I think that this indirection makes less sense when the method is short and when it is used in relatively few places.